src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -276,6 +277,11 @@ protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder
                 }
             }
 
+            if (this.IsNullChecked && convertedExpression.ConstantValue?.IsNull == true)
+            {
+                diagnostics.Add(ErrorCode.WRN_NullCheckedHasDefaultNull, Locations.FirstOrNone(), this);
```

Comment:
>this [](start = 98, length = 4)

`Name`

Reply:
It looks like the existing implementation produces diagnostics like the following:

```cs
// (4,19): warning CS8719: Parameter 'string' is null-checked but is null by default.
//     void M(string name! = null) { }
Diagnostic(ErrorCode.WRN_NullCheckedHasDefaultNull, 
```

I'm slightly surprised by this. But it makes some sense in the context of converting an entire MethodSymbol to a display string.